src/replica/replica_stub.cpp

```diff
@@ -2751,6 +2740,77 @@ void replica_stub::update_disk_holding_replicas()
     }
 }
 
+void replica_stub::gc_disk_replica_folder()
```

Comment:
I prefer making this routine a pure function:

disk_cleaner.h

```cpp
struct disk_cleaning_report
{
  int garbage_replica_count;
  int error_replica_count;
  int disk_migrate_tmp_count;
  int disk_migrate_origin_count;
};

/// Removes the useless data from data directories.
error_s disk_remove_useless_dirs(const std::vector<std::string&> data_dirs, /*output*/ disk_cleaning_report &report);
```

1. This design is extensible for future possible disk cleaning strategies.
2. This design decouples disk cleaning with `replica_stub`. I think they have nothing relevant essentially.
3. The function is testable even without setting up a `replica_stub`.

Reply:
Right,  is the change large in this pr?